xcube_server/controllers/time_series.py

```diff
@@ -18,3 +18,65 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
+
+from ..context import ServiceContext
+from ..errors import ServiceBadRequestError
+from ..reqparams import RequestParams
+import numpy as np
+from pandas import Timestamp
+from typing import Dict
+
+
+def get_time_series_info(ctx: ServiceContext) -> Dict:
+    time_series_info = {'layers': []}
+    descriptors = ctx.get_dataset_descriptors()
+    for descriptor in descriptors:
+        if 'Identifier' in descriptor:
+            dataset = ctx.get_dataset(descriptor['Identifier'])
+            if 'time' not in dataset.variables:
+                continue
+            attributes = dataset.attrs
+            bounds = {'xmin': attributes.get('geospatial_lon_min', -180),
+                      'ymin': attributes.get('geospatial_lat_min', -90),
+                      'xmax': (attributes.get('geospatial_lon_max', +180)),
+                      'ymax': attributes.get('geospatial_lat_max', +90)}
+            time_data = dataset.variables['time'].data
+            time_stamps = []
+            for time in time_data:
+                time_stamps.append(Timestamp(time).strftime('%Y-%m-%dT%H:%M:%SZ'))
+            for variable in dataset.data_vars.variables:
+                variable_dict = {'name': '{0}.{1}'.format(descriptor['Identifier'], variable),
+                                 'dates': time_stamps,
+                                 'bounds': bounds}
+                time_series_info['layers'].append(variable_dict)
+    return time_series_info
+
+
+def get_time_series_for_point(ctx: ServiceContext, ds_name: str, var_name: str, params: RequestParams) -> Dict:
+    lat = params.get_query_argument_float('lat', default=None)
+    lon = params.get_query_argument_float('lon', default=None)
+    if lat is None or lon is None:
+        raise ServiceBadRequestError('lat and lon must be given as query parameters')
+    start_date = np.datetime64(params.get_query_argument('startDate', default='1970-01-01'))
+    end_date = np.datetime64(params.get_query_argument('endDate', default='2099-12-31'))
+    dataset, variable = ctx.get_dataset_and_variable(ds_name, var_name)
+    dim_names = list(variable.dims)
+    if 'lon' not in dim_names or 'lat' not in dim_names:
+        raise ServiceBadRequestError(f'variable {var_name!r} of dataset {ds_name!r} is not geo-spatial')
+    if 'time' not in dim_names:
+        raise ServiceBadRequestError(f'variable {var_name!r} of dataset {ds_name!r} has no time information')
+    point_subset = variable.sel(lat=lat, lon=lon, method='Nearest')
```

Comment:
Hier gibt es ein Problem: wenn der lon/lat point gar nicht in den Dataset-Bounds vorkommt, wird dur `sel()` trotzdem der "nearest" Point zurückgegeben. Das müssen wir noch abfangen und dann einen leeren Response zurückgeben.